pkg/kubecost/range.go

```diff
@@ -0,0 +1,73 @@
+package kubecost
+
+import (
+	"fmt"
+	"github.com/kubecost/cost-model/pkg/util/json"
+	"sync"
+)
+
+// Range is a generic implementation
+type Range[T Set] struct {
```

Comment:
`ETLRange`? I'm more ok with just `Range` here than I am with `Set` so I'll leave it up to you.